### ISSUE TITLE
Schemas added with GenerateSchema are not included in swagger.json

### ISSUE THREAD

**Issue by ErikPilsits-RJW** [domaindrivendev/Swashbuckle.AspNetCore#2000]
I have the following document filter which adds schemas which are not directly used in my controller methods, but which can be part of dynamic request payloads.
```csharp
public class ExtraTypesDocumentFilter : IDocumentFilter
{
    public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
    {
        var types = AppDomain.CurrentDomain.GetAssemblies()
            .SelectMany(t => t.GetTypes()).Where(t =>
                t.IsClass
                && t.Namespace.InsensitiveEquals("MyNamespace.Models")
                && !(t.BaseType?.Name.InsensitiveEquals("RowBase") ?? false));

        foreach (var typ in types)
            context.SchemaGenerator.GenerateSchema(typ, context.SchemaRepository);
    }
}
```
If I put a break at the end of this method, I can see the target types have successfully been added to the SchemaRepository.

This worked through 6.0.1, then in 6.0.2 it stopped working and the desired models are no longer added to my swagger.json. I think the issue is in the Swashbuckle.AspNetCore.SwaggerGen package, but I can't be sure. I'm using the main package, Annotations and Newtonsoft packages. If I upgrade any of the addons to 6.0.2, for example Newtonsoft, it brings in the SwaggerGen 6.0.2 dependency and the correct swagger.json is no longer generated.

**Comment by domaindrivendev** [comment]
Hmmm - this is not good, looking into it and will get a hotfix out ASAP.

**Comment by martincostello** [comment]
Had a similar issue with this, except something I'd removed was still being returned. 6.0.3 fixes it.

**Comment by ErikPilsits-RJW** [comment]
Version 6.0.3 also seems to fix my issue. @domaindrivendev should I consider 6.0.3 the official fix that I can roll out, or are you still doing investigation?

**Comment by domaindrivendev** [comment]
Issue fixed as of `6.0.3`
